router/context.ts

```diff
@@ -73,7 +73,12 @@ export type ProcedureHandlerContext<State> = ServiceContext & {
   /**
    * This signal is a standard [AbortSignal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal)
    * triggered when the procedure invocation is done. This signal tracks the invocation/request finishing
-   * for _any_ reason including the procedure invocation.
+   * for _any_ reason, for example:
+   * - client explicit cancellation
+   * - procedure handler explicit cancellation via {@link cancel}
+   * - client session disconnect
+   * - server cancellation due to client invalid payload
+   * - invocation finishing cleanly (i.e. rpc handler return, or both sides closing streams)
```

Comment:
```suggestion
   * - invocation finishing cleanly (i.e. rpc handler return, or either sides closing streams)
```

from my understanding, the signal happens when either side closes the stream. if it doesn't, might be worth adding a clarification.

Reply:
No, it's both closing their end. Half-close doesn't trigger the signal. I'll add clarification.

Reply:
maybe this?

`invocation finishing cleanly (i.e. rpc handler return, or after the client-side has ended the writable and the server-side has ended the readable)`